### PR TITLE
Added DNS Server APIs

### PR DESCRIPTION
**Pull request by ragupta-git**
Added DNS Server APIs

### CODE REVIEW

File: ucsm_apis/admin/dns.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module performs the operation related to dns server management.
+"""
+from ucsmsdk.ucsexception import UcsOperationError
+from ..common.utils import get_device_profile_dn
+ucsc_base_dn = get_device_profile_dn(name="default")
```
Comment (vvb) [inline ucsm_apis/admin/dns.py]: ucsc_base_dn ?

File: ucsm_apis/admin/dns.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module performs the operation related to dns server management.
+"""
+from ucsmsdk.ucsexception import UcsOperationError
+from ..common.utils import get_device_profile_dn
```
Comment (vvb) [inline ucsm_apis/admin/dns.py]: where and what is `get_device_profile_dn` ?

File: ucsm_apis/admin/dns.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+ucsc_base_dn = get_device_profile_dn(name="default")
+
+_dns_server_parent_dn = "sys/svc-ext/dns-svc"
```
Comment (vvb) [inline ucsm_apis/admin/dns.py]: `_dns_server_parent_dn` -> `_dns_svc_dn`

File: ucsm_apis/admin/dns.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+def dns_server_get(handle, name, caller="dns_server_get"):
```
Comment (vvb) [inline ucsm_apis/admin/dns.py]: `caller` is not required here. It was added in boot APIs as we were overloading calls of a method from multiple internal methods and wanted a way to print that info. This is not the case here.

Comment (ragupta-git) [inline ucsm_apis/admin/dns.py]: Here also we are calling `dns_server_get` in modify, delete and exist API, similar to boot one.

File: ucsm_apis/admin/dns.py (hunk 1 above)
Comment (ragupta-git) [inline ucsm_apis/admin/dns.py]: Removed.

File: ucsm_apis/admin/dns.py (hunk 2 above)
Comment (ragupta-git) [inline ucsm_apis/admin/dns.py]: Removed.

File: ucsm_apis/admin/dns.py (hunk 3 above)
Comment (ragupta-git) [inline ucsm_apis/admin/dns.py]: changed.

File: ucsm_apis/admin/dns.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+def dns_server_exists(handle, name, **kwargs):
+    """
+    Checks if the dns entry already exists
+
+    Args:
+        handle (UcsHandle)
+        name (string): IP address of the dns server
+        **kwargs: key-value pair of managed object(MO) property and value, Use
+                  'print(ucscoreutils.get_meta_info(<classid>).config_props)'
```
Comment (vvb) [inline ucsm_apis/admin/dns.py]: ucsc -> ucs

Comment (ragupta-git) [inline ucsm_apis/admin/dns.py]: `ucsccoreutils` changed to `ucscoreutils`.

**Pull request merged**